### PR TITLE
Separate Custom Error Handling.

### LINKED ISSUE #60: [Bug]: CustomError should have a handleError()
### Preflight Checklist

- [X] I have read the Contribution.md for this project.
- [X] I agree to follow the Code of Conduct that this project adheres to.
- [X] I have searched the issue for a feature request that matches the one I want to file, without success.

### Version

0.0.1

### Current Behavior

We are handling response for `CustomError` in the `errorhandler.helper.js` so in future if we multiple `CustomError` which needs to be handled differently then we will had to increase the if-else ladder in `errorhandler.helper.js`.

### Expected Behavior

`CustomError` should have a `handleError` which will take `res` as parameter and abstracts :

```js
res.status(err.status).json({
   error: err.getResponse()
});
```

And this `err.handleError(res)` should be called in  `errorhandler.helper.js`.

### Additional Information

NA

### PR DESCRIPTION
**Pull request by Vraj9799**
# Description:

I have separated the custom error handling from the Error Handler to CustomError.

Fixes #60 

## Type of change:

-   [X] Bug fix (non-breaking change which fixes an issue)

# Checklist:

-   [x] My code follows the style guidelines of this project.
-   [x] I have performed a self-review of my own code.
-   [x] My changes generate no new warnings.

### CODE REVIEW

**Comment by Vraj9799** [comment]
Hi @Shubham-Kumar-2000 I created a PR please verify the change.

File: src/helpers/errorhandler.helper.js [hunk 1]
```diff
@@ -4,9 +4,7 @@ const CustomError = require('../errors/custom.error');
 // eslint-disable-next-line prettier/prettier, no-unused-vars
 exports.errorHandler = (err, req, res, _) => {
     if (err instanceof CustomError) {
-        return res.status(err.status).json({
-            error: err.getResponse()
-        });
+        err.handleError(res);
```
Comment (Shubham-Kumar-2000) [inline src/helpers/errorhandler.helper.js]: Return it from here.

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]
Please run test cases before commit 
By running :
```bash
npm test
```

**Comment by Shubham-Kumar-2000** [comment]
@VRAJ9799  please make necessary changes.

**Comment by Vraj9799** [comment]
Hi @Shubham-Kumar-2000 Can you please guide me about this error "const isDevelopment = req.app.get('env') === 'development';" i haven't changed anything here.

**Comment by Shubham-Kumar-2000** [comment]
@VRAJ9799  Join here https://meet.google.com/yun-diwh-whe

**Comment by Shubham-Kumar-2000** [comment]
Hi @VRAJ9799  I am dropping off please comment your available time. Will send a meet link at that time.

**Comment by Vraj9799** [comment]
Hi @Shubham-Kumar-2000 I have push the changes.

**Review by Shubham-Kumar-2000 — APPROVED** [review]

**Comment by Vraj9799** [comment]
Hi @Shubham-Kumar-2000 Can you please add gssoc tag so this pr can count for that ?

**Comment by Shubham-Kumar-2000** [comment]
Done @VRAJ9799  you can try other open issues.

**Pull request merged**